ondevice_learning.qmd

```diff
@@ -1,61 +1,630 @@
+
 ::: {.callout-tip}
 ## Learning Objectives
 
-* coming soon.
+* Understand on-device learning and how it differs from cloud-based training
+
+* Recognize the benefits and limitations of on-device learning
+
+* Examine strategies to adapt models through complexity reduction, optimization, and data compression 
+
+* Understand related concepts like federated learning and transfer learning
+
+* Analyze the security implications of on-device learning and mitigation strategies
 
 :::
 
+On-device Learning represents a significant innovation for embedded and edge IoT devices, enabling models to train and update directly on small local devices. This contrasts with traditional methods where models are trained on expansive cloud computing resources before deployment. With On-Device Learning, devices like smart speakers, wearables, and industrial sensors can refine models in real-time based on local data, without needing to transmit data externally. For example, a voice-enabled smart speaker could learn and adapt to its owner's speech patterns and vocabulary right on the device. But there is no such thing as free lunch, therefore in this chapter, we will discuss both the benefits and the limitations of on-device learning. 
+
 ## Introduction
 
-Explanation: This section sets the stage for the reader, explaining why on-device learning is a critical aspect of embedded AI systems.
+On-device Learning refers to the process of training machine learning models directly on the device where they are deployed, as opposed to traditional methods where models are trained on powerful servers and then deployed to devices. This method is particularly relevant to TinyML, where machine learning systems are integrated into tiny, resource-constrained devices.
+
+An example of On-Device Learning can be seen in a smart thermostat that adapts to user behavior over time. Initially, the thermostat may have a generic model that understands basic patterns of usage. However, as it is exposed to more data, such as the times the user is home or away, preferred temperatures, and external weather conditions, the thermostat can refine its model directly on the device to provide a personalized experience for the user. This is all done without the need to send data back to a central server for processing. 
+
+Another example is in predictive text on smartphones. As users type, the phone learns from the user’s language patterns and suggests words or phrases that are likely to be used next. This learning happens directly on the device, and the model updates in real-time as more data is collected. A widely used real-world example of on-device learning is Gboard. On an Android phone, the Gboard keyboard can learn from your keyboard and dictation use to help improve Gboard for everyone. 
```

Comment:
maybe change "On an Android phone, the Gboard keyboard can learn from your keyboard and dictation use to help improve Gboard for everyone." to "On an Android phone, Gboard learns from typing and dictation patterns to enhance the experience for all users."

Reply:
Thanks!